org.scala-ide.sdt.core/src/org/scalaide/core/internal/builder/zinc/SbtInputs.scala

```diff
@@ -53,11 +59,21 @@ class SbtInputs(installation: IScalaInstallation,
   def analysisMap(f: File): Maybe[Analysis] =
     if (f.isFile)
       Maybe.just(Analysis.Empty)
-    else
-      allProjects.find(_.sourceOutputFolders.map(_._2.getLocation().toFile()).toSeq contains f) map (_.buildManager) match {
-        case Some(sbtManager: EclipseSbtBuildManager) => Maybe.just(sbtManager.latestAnalysis(incOptions))
-        case _                                     => Maybe.just(Analysis.Empty)
+    else {
+      def suchProjectWhereFileIsProduct(project: IScalaProject): Boolean = {
+        val outputFolders = project.sourceOutputFolders map {
+          case (_, outputFolder) => outputFolder.getLocation.toFile
+        }
+        outputFolders contains f
+      }
+
+      val backingManager: IScalaProject => EclipseBuildManager = _.buildManager
+      allProjects find suchProjectWhereFileIsProduct map backingManager match {
```

Comment:
sorry, the name has left from previous code where on manager the method called getInner to retrieve the working manager. Now is not needed